File: arrays/Geeks/com/CheckSorted.java
package Geeks.com;

public class CheckSorted {
    public static void main(String[] args) {
        int[] arr = {7,20, 30, 10};
        System.out.println(isSorted(arr));
        System.out.println(efficientApproach(arr));
    }

    public static boolean isSorted(int[] arr){
        for (int j : arr) {
            for (int k : arr) {
                if (k < j) {
                    return false;
                }
            }
        }
        /*
        for(int i=0; i<arr.length; i++){
            for (int j = 0; j < arr.length; j++) {
                if(arr[j] < arr[i]){
                    return false;
                }
            }
        }*/
        return true;
    }

    public static boolean efficientApproach(int[] arr) {
        for (int i = 1; i < arr.length; i++) {
            if(arr[i] < arr[i-1]){
                return false;
            }
        }
        return true;
    }
}

File: P5. Arrays/takeuforward/CheckSorted.cpp

#include <bits/stdc++.h>
using namespace std;



bool isSorted(int arr[], int n) {
  for (int i = 0; i < n; i++) {
    for (int j = i + 1; j < n; j++) {
      if (arr[j] < arr[i])
        return false;
    }
  }

  return true;
}



bool isSortedNot(int arr[], int n) {
  for (int i = 1; i < n; i++) {
    if (arr[i] < arr[i - 1])
      return false;
  }

  return true;
}


int main() {

  int arr[] = {1, 2, 3, 4, 5}, n = 5;
  bool ans = isSorted(arr, n);
  if (ans) cout << "True" << endl;
  else cout << "False" << endl;


  int arr1[] = {1, 2, 3, 4, 5}, n = 5;
  printf("%s", isSortedNot(arr1, n) ? "True" : "False");

  return 0;
}

